core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -776,16 +801,55 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
     val resourceProfileToNumExecutors = resourceProfileIdToNumExecutors.map { case (rpid, num) =>
       (scheduler.sc.resourceProfileManager.resourceProfileFromId(rpid), num)
     }
+    val oldResourceProfileToNumExecutors = requestedTotalExecutorsPerResourceProfile.map {
+      case (rp, num) =>
+        (rp.id, num)
+    }.toMap
```

Comment:
Move this into the `synchronized` block ?

Reply:
Good point